README.md
# 普药险项目

## Project setup
```
npm install
```

### Compiles and hot-reloads for development
```
npm run serve
```

### Compiles and minifies for production
```
npm run build
```

### Lints and fixes files
```
npm run lint
```

### Customize configuration
See [Configuration Reference](https://cli.vuejs.org/config/).


changeLog.md
# 普药险项目

## 2021-11-10首次提交代码
```

```
## 2011-11-10 修复bug

src/main.js
import { createApp } from 'vue'
import App from './App.vue'
import router from './router'
import axios from 'axios'
import store from '@/store'

import { DatetimePicker } from 'vant';
import { Overlay } from 'vant';
import { Loading } from 'vant';
import lazyPlugin from 'vue3-lazy'
import Vconsole from 'vconsole'
// 待会注释
// let vConsole = new Vconsole();

const setHtmlFontSize = () => {
  const htmlDom = document.getElementsByTagName('html')[0];
  let htmlWidth = document.documentElement.clientWidth || document.body.clientWidth;
  if (htmlWidth >= 750) {
    htmlWidth = 750;
  }
  if (htmlWidth <= 320) {
    htmlWidth = 320;
  }
  //htmlDom.style.fontSize = `${htmlWidth / 7.5}px`;
  htmlDom.style.fontSize = `20px`;
};
window.onresize = setHtmlFontSize;
setHtmlFontSize();

const app = createApp(App);
app.config.globalProperties.$axios = axios
// app.use(Vant);
app.directive('throttle', {
  bind: (el, binding) => {
    let throttleTime = binding.value; // 防抖时间
    if (!throttleTime) { // 用户若不设置防抖时间，则默认2s
      throttleTime = 2000;
    }
    let cbFun;
    el.addEventListener('click', event => {
      if (!cbFun) { // 第一次执行
        cbFun = setTimeout(() => {
          cbFun = null;
        }, throttleTime);
      } else {
        event && event.stopImmediatePropagation();
      }
    }, true);
  },
});

app
// 待会注释
// .use(vConsole)

.use(lazyPlugin, {
  loading: 'loading.png',
  error: 'error.png'
})
.use(Overlay)
.use(DatetimePicker)
.use(Loading)
.use(router)
.use(store)
.mount('#app')


vue.config.js
const path = require('path');
module.exports = {
  publicPath: './',
  outputDir: 'hm',
  css: {
    loaderOptions: {
      //  sass: {
      //      additionalData:`@import "./src/assets/scss/style.scss";` 
      //      }
       }
   },
   devServer: {
    port: 8080,
    proxy: {
      '/api': {
       target: 'http://120.221.160.193:9009/',
      //  'target': 'http://10.89.7.229:9010/',
        // target : 'http://10.89.7.111:9006/',
        ws: true,
        changeOrigin: true,
        logLevel: 'debug',
        pathRewrite: {
          '^/api': ''
        }
      }
    }
  },
  lintOnSave: false,// 屏蔽EsLint,
  transpileDependencies:['@vue/reactivity'],
  chainWebpack: config => {
      // ...
      // config.plugins.delete('prefetch')

      config.module.rule('pdfjs-dist').test({
        test: /\.js$/,
        include: path.join(__dirname, 'node_modules/pdfjs-dist')
      }).use('babel-loader').loader('babel-loader').options({
        presets: ['@babel/preset-env'],
        plugins: ['@babel/plugin-proposal-optional-chaining']
      })
      config.module
      .rule('pdf')
      .test(/\.(pdf)$/)
      .use('url-loader')
      .loader('url-loader')
      .options({
        limit: 10000
      })
      .end()
  }
}


src/utils/common.js
export function getQueryParam(str) {
  if(str) {
    let link = window.location.href;
    let query = link.split('?');
    if(query && query.length>1){
      let queryArray = query[1].split('&');
      if(queryArray && queryArray.length>0){
        for(let i=0, len= queryArray.length; i<len; i++){
          let params = queryArray[i].split('=');
          if(params && params.length>0 && params[0].toLowerCase() === str.toLowerCase()){
            return params[1];
          }
        }
      }
    }
  } else {
    return ''
  }
}

export function preventCopy() {
   let link = window.location.href;
   let query = link.split('?');
   let theRequest = new Object();
   if (query && query.length>1) {
      let str = query[1];
      let strs = str.split("&");
      for(let i = 0; i < strs.length; i ++) {
        if(strs[i].toLowerCase().indexOf('sdk')!=-1){
          theRequest.sdkProductId = strs[i].slice(13,strs[i].length);
        }
        if(strs[i].toLowerCase().indexOf('user')!=-1){
          theRequest.userId = strs[i].slice(7,strs[i].length);
        }
        if(strs[i].toLowerCase().indexOf('ten')!=-1){
          theRequest.tenantId = strs[i].slice(9,strs[i].length);
        }
        theRequest.link = query[0];
        console.log(theRequest);
        // theRequest[strs[i].split("=")[0]]=unescape(strs[i].split("=")[1]);
      }
      return theRequest;
   } else {
     return {
       link: query[0]
     }
   }
   
}
export function preventCopy1() {
    let link = window.location.href;
    let query = link.split('?');

    if(query && query.length>1){
      
      let returnObj = {};
      let queryArray = query[1].split('&');
      if(queryArray && queryArray.length>0){
        returnObj.link = query[0];
        for(let i=0, len= queryArray.length; i<len; i++){
          let params = queryArray[i].split('=');
          if(params && params.length>0 && params[0].toLowerCase() === 'sdkProductId'.toLowerCase()){
            returnObj.sdkProductId=params[1];
          }
          if(params && params.length>0 && params[0].toLowerCase() === 'userId'.toLowerCase()){
            returnObj.userId=params[1];
          }
          if(params && params.length>0 && params[0].toLowerCase() === 'tenantId'.toLowerCase()){
            returnObj.tenantId=params[1];
          }
          
        }
        // let queryStr =new URLSearchParams('?'+query[1]);
        // returnObj.tenantId = queryStr.get('tenantid') || queryStr.get('tenantId');
        // returnObj.userId = queryStr.get('userId');
        // returnObj.sdkProductId = queryStr.get('sdkProductId');
        // console.log(returnObj , 333);
        return returnObj;
      }
    }else{
      return {
        link: link
      }
    }
}

export function getNumber(num){
  return num<10 ? `0${num}`: num;
}

export function NorEmpty(str){
  if(str && str!=null && str!='null'){
    return true;
  }else 
    return false;
    
}

export function desensitizationName(name){
  if(!NorEmpty(name)){
    return name
  } else if(name.length<=1){
    return name
  } else if(name.length==2){
    return `${name.charAt(0)}*`
  }else {
    let str='';
    for(let i=0,len=name.length-2; i<len; i++){
      str +='*';
    }
    return `${name.charAt(0)}${str}${name.charAt(name.length-1)}`;
  }
}

export function desensitizationCard(card){
  if(card.match(/^[1-9]\d{5}(?:18|19|20)\d{2}(?:0[1-9]|10|11|12)(?:0[1-9]|[1-2]\d|30|31)\d{3}[\dX]$/)) {
    // console.log(card.match(/^[1-9]\d{5}(?:18|19|20)\d{2}(?:0[1-9]|10|11|12)(?:0[1-9]|[1-2]\d|30|31)\d{3}[\dX]$/),'身份证');
    let cardText='**************';
    return `${cardText}${card.slice(14,18)}`
  } else {
    return '';
  }
}

src/utils/index.js
// const colorArray=['aliceblue', 'antiquewhite', 'aqua', 'aquamarine', 'azure', 'beige',
// 	'bisque', 'black', 'blancedalmond', 'blue','blueviolet', 'brown', 'burlywood',
// 	 'cadetblue', 'chartreuse', 'chocolate', 'coral', 'cornflowerblue', 'cornsilk', 
// 	  'cyan', 'darkblue', 'darkcyan', 'darkgoldenrod', 'darkgray', 'darkgreen', 
// 		 'darkkhaki', 'darkmegenta', 'darkolivegreen', 'darkorange', 'darkorchid',
// 			'darkred', 'darksalmon', 'darkseagreen', 'darkslateblue', 'darkslategray', 'darkturquoise',
// 			 'darkviolet', 'deeppink', 'deepskyblue', 'dimgray', 'dodgeblue' ];
//判断引用类型的具体类型
export function objectType(variable) {
	var obj = Object.prototype.toString;
	var type = obj.call(variable);
	switch (type) {
		case '[object Array]':
			return 'array';
			break;
		case '[object String]':
			return 'string';
			break;
		case '[object Object]':
			return 'object';
			break;
		case '[object Null]':
			return 'null';
			break;
		case '[object Undefined]':
			return 'undefined';
			break;
		case '[object Boolean]':
			return 'boolean';
			break;
		case '[object Number]':
			return 'number';
			break;
		case '[object Function]':
			return 'function';
			break;
		case '[object Date]':
			return 'date';
			break;
		default:
			break;
	}
}

/***********/
//快速排序
export function quickSort(arr, sortOrder = 'asc') {
	if (objectType(arr) != 'array') {
		return false;
	} else {
		if (arr.length === 1 || arr.length === 0) {
			return arr;
		} else {
			var symbol = arr.splice(0, 1);
			var left = [];
			var right = [];
			if (sortOrder == 'asc') {
				arr.forEach((item) => {
					if (item < symbol)
						left.push(item);
					else
						right.push(item);
				});
			} else if (sortOrder == 'desc') {
				arr.forEach((item) => {
					if (item < symbol)
						right.push(item);
					else
						left.push(item);
				});
			}
			return quicksort(left, sortOrder).concat(symbol, ...quicksort(right, sortOrder));
		}
	}
}
//归并排序
export function mergeSort(array) {
	if (array.length == 1) {
		return array;
	}
	let m = Math.floor(array.length / 2);
	let B = mergeSort(array.slice(0, m));
	let C = mergeSort(array.slice(m, array.length));
	let A_sorted = merge(B, C);
	return A_sorted;
}

function merge(B, C) {
	let i = 0;
	let j = 0;
	let D = [];
	while (i < B.length && j < C.length) {
		if (B[i] < C[j]) {
			D.push(B[i++])
		} else {
			D.push(C[j++]);
		}
	}
	while (i < B.length)
		D.push(B[i++]);
	while (j < C.length)
		D.push(C[j++])
	return D;
}

/*****排序结束******/

/*****查找******/

/*****查找结束******/

/*****实现深拷贝******/

// lmran
function copyFunction(func) {
	let fnStr = func.toString()
	return func.prototype ? eval(`(${fnStr})`) : eval(fnStr)
}

export function deepCopy(obj, cache = []) {
	if (typeof obj === 'function') {
		return copyFunction(obj)
	}
	if (obj === null || typeof obj !== 'object') {
		return obj
	}

	if (Object.prototype.toString.call(obj) === '[object Date]') return new Date(obj)
	if (Object.prototype.toString.call(obj) === '[object RegExp]') return new RegExp(obj)
	if (Object.prototype.toString.call(obj) === '[object Error]') return new Error(obj)

	const item = cache.filter(item => item.original === obj)[0]
	if (item) return item.copy

	let copy = Array.isArray(obj) ? [] : {}
	cache.push({
		original: obj,
		copy
	})

	Object.keys(obj).forEach(key => {
		copy[key] = deepCopy(obj[key], cache)
	})

	return copy
}

/*****数组扁平化******/

export function flattenArr(arr) {
	let result = [];
	arr.forEach((item, index, arr) => {
		if (Array.isArray(item)) {
			result = result.concat(flatten(item))
		} else {
			result.push(item);
		}
	})
	return result;
}

/*****常用正则******/

export function isEmail(str) { //是否是邮箱
	//Email正则
	var emailPattern = /^([A-Za-z0-9_\-\.])+\@([A-Za-z0-9_\-\.])+\.([A-Za-z]{2,4})$/;
	//输出 true
	return !str ? false : emailPattern.test(str);
}


export function isPhone(str) { //是否是手机号
	str.substring(str.length-11)
	var phonePattern = /^(1[3-9]\d{9})$/;
	return !str ? false : phonePattern.test(str);
}

export function isIdCard(str) { //是否是身份证号
	var IdCardPattern =
		/^[1-9]\d{5}(18|19|([23]\d))\d{2}((0[[1-9])|(10)|(11)|(12))(([0-2][1-9])|(10)|(20)|(30)|(31))\d{3}[0-9Xx]$/
	return !str ? false : IdCardPattern.test(str)
}

export function isWeChat(str) { //是否微信号
	var weChatPattern = /^[a-zA-Z]([-_0-9a-zA-Z]{5,19})+$/;
	return !str ? false : weChatPattern.test(str);
}

export function isQQ(str) { //是否是QQ
	var QQPattern = /^[1-9][0-9]{4,10}$/;
	return !str ? false : QQPattern.test(str);
}

export function isPlateNumber(str) { //是否是车牌号
	var carPattern = /^[京津沪渝冀豫云辽黑湘皖鲁新苏浙赣鄂桂甘晋蒙陕吉闽贵粤青藏川宁琼使领A-Z]{1}[A-Z]{1}[A-Z0-9]{4}[A-Z0-9挂学警港澳]{1}$/;
	return !str ? false : carPattern.test(str);
}

export function containCN(str) { //是否包含中文字符
	var cnPattern = /[\u4E00-\u9FA5]/;
	return !str ? false : cnPattern.test(str);
}

export function UpperMoney(money) { //人民币转换中文表示
	//数字
	let cnNums = new Array('零', '壹', '贰', '叁', '肆', '伍', '陆', '柒', '捌', '玖');
	//基本单位
	let cnIntRadice = new Array('', '拾', '佰', '仟');
	//对应整数部分扩展单位
	let cnIntUnits = new Array('', '万', '亿', '兆');
	//对应小数部分
	let cnDecUnits = new Array('角', '分', '毫', '厘');
	//整数金额后面跟的字符
	let cnInteger = '整';
	//整数完以后的单位
	let cnIntLast = '元';
	//整数部分
	let integerNum;
	//小数部分
	let decimalNum;
	//结果字符串
	let CNstr = '';
	if (isNaN(Number(money))) {
		return '';
	}
	if (money == 0) {
		CNstr = cnNums[0] + cnIntLast + cnInteger;
		return CNstr;
	}
	money = money.toString();
	if (money.indexOf('.') == -1) {
		integerNum = money;
		decimalNum = '';
	} else {
		let parts = money.split('.');
		integerNum = parts[0];
		decimalNum = parts[1].substr(0, 4);
	}
	if (parseInt(integerNum, 10) > 0) {
		var zeroCount = 0;
		var IntLen = integerNum.length;
		for (var i = 0; i < IntLen; i++) {
			var n = integerNum.substr(i, 1);
			var p = IntLen - i - 1;
			var q = p / 4;
			var m = p % 4;
			if (n == '0') {
				zeroCount++;
			} else {
				if (zeroCount > 0) {
					CNstr += cnNums[0];
				}
				//归零
				zeroCount = 0;
				CNstr += cnNums[parseInt(n)] + cnIntRadice[m];
			}
			if (m == 0 && zeroCount < 4) {
				CNstr += cnIntUnits[q];
			}
		}
		CNstr += cnIntLast;
	}
	//小数部分
	if (decimalNum != '') {
		var decLen = decimalNum.length;
		for (var i = 0; i < decLen; i++) {
			var n = decimalNum.substr(i, 1);
			if (n != '0') {
				CNstr += cnNums[Number(n)] + cnDecUnits[i];
			}
		}
	}
	if (CNstr == '') {
		CNstr += cnNums[0] + cnIntLast + cnInteger;
	} else if (decimalNum == '') {
		CNstr += cnInteger;
	}
	return CNstr;
}

// export function isColor(str){ //是否正确颜色代码
// 	var str=str.trim();
// 	var rgbPattern = /^rgb\(\s*(\d+)\s*\,\s*(\d+)\s*\,\s*(\d+)\s*\)$/;
// 	var hexPattern = /^#[0-9a-fA-F]{6}$/;
// 	var hexPattern2 = /^#[0-9a-fA-F]{3}$/;
// 	if(rgbPattern.test(str)){
// 		var array=str.match(rgbPattern);
// 		if(array){
// 			for(var i=1;i<4;i++){
// 				if(array[i]<0||array[i]>255){
// 					return false;
// 				}
// 			}
// 			return true;
// 		}else{
// 			return false;
// 		}		
// 	}else if(hexPattern.test(str) || hexPattern2.test(str)){
// 		return true;
// 	}else if(str.toLowerCase() in colorArray){
// 		return true;
// 	}else{
// 		return false;
// 	}
// }
/*****正则结束******/

/*****格式化日期时间******/

function getTime(str) {
	let time;
	if (str instanceof Date == true) {
		time = str;
	} else {
		time = new Date(str);
	}
	return time;
}

export function transforNumber(num) {
	return num > 9 ? num : '0' + num;
}

export function formatterDate(str = Number(new Date().valueOf()), concat) { //只获取日期
	let time = getTime(str);
	let year = time.getFullYear();
	let month = transforNumber(time.getMonth() + 1);
	let date = transforNumber(time.getDate());
	concat = concat || '-';
	let result = year + concat + month + concat + date;
	return result;
}

export function formatterTime(str = Number(new Date().valueOf())) {
	let time = getTime(str);
	let hour = time.getHours();
	let minutes = transforNumber(time.getMinutes());
	let seconds = transforNumber(time.getSeconds());
	let result = hour + ':' + minutes + ':' + seconds;
	return result;
}

export function formatterDateTime(str = Number(new Date().valueOf())) { //获取日期和时间
	var result = formatterDate(str) + ' ' + formatterTime(str);
	return result;
}

/*****格式化日期时间******/

/*****是否支持webp******/

//获取ios版本
function get_ios_version(){
	var ua = navigator.userAgent.toLowerCase();
	var version = null;
	if (ua.indexOf("like mac os x") > 0) {
			var reg = /os [\d._]+/gi;
			var v_info = ua.match(reg);
			version = (v_info + "").replace(/[^0-9|_.]/ig, "").replace(/_/ig, "."); //得到版本号9.3.2或者9.0
			version = parseInt(version.split('.')[0]); // 得到版本号第一位
	}

	return version;
}

export function check_webp_feature(feature, callback) {
	var kTestImages = {
			lossy: "UklGRiIAAABXRUJQVlA4IBYAAAAwAQCdASoBAAEADsD+JaQAA3AAAAAA",
			lossless: "UklGRhoAAABXRUJQVlA4TA0AAAAvAAAAEAcQERGIiP4HAA==",
			alpha: "UklGRkoAAABXRUJQVlA4WAoAAAAQAAAAAAAAAAAAQUxQSAwAAAARBxAR/Q9ERP8DAABWUDggGAAAABQBAJ0BKgEAAQAAAP4AAA3AAP7mtQAAAA==",
			animation: "UklGRlIAAABXRUJQVlA4WAoAAAASAAAAAAAAAAAAQU5JTQYAAAD/AABBTk1GJgAAAAAAAAAAAAAAAAAAAGQAAABWUDhMDQAAAC8AAAAQBxAREYiI/gcA"
	};
	var img = new Image();
	img.onload = function () {
			var result = (img.width > 0) && (img.height > 0);
			callback(feature, result);
	};
	img.onerror = function () {
			callback(feature, false);
	};
	img.src = "data:image/webp;base64," + kTestImages[feature];
}

export function getWebp(){
	return new Promise(function(resolve){
		check_webp_feature('lossy', (feature,result)=>{
			resolve(result);
		});
	})
}



/*****是否支持webp结束******/


